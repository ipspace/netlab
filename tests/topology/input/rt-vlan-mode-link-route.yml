# This is a regression test for #881 that tries to address all possible combinations
# of link-, node- and vlan forwarding modes, in particular with mode = route
#

defaults.device: frr
module: [ vlan ]

vlans:
  red:
    mode: route           # red VLAN is primarily routed
  blue:
    mode: bridge          # blue one is primarily bridged
  green:                  # and green is IRB

nodes:
  h:
    device: linux         # A host -- needed to check we don't set _vlan_mode on host interfaces
  rt:
  sw:
    vlans:
      red:
        mode: bridge      # Change a routed VLAN into bridged one
  br:
    vlan.mode: bridge     # And set the default VLAN mode to bridge

links:
- h:                      # Test that the _vlan_mode setting does not apply to host interface
  rt:                     # Should be a routed link
  vlan.access: red
  name: Host access link
#  disable: True
- h:                      # Should be a bridged VLAN
  sw:
  vlan.access: red
  name: Host access link to a switch
#  disable: True
- h:                      # Should be a bridged VLAN
  br:
  vlan.access: red
  name: Host access link to a bridge
#  disable: True
- sw:
  br:
  vlan.trunk:
    red:                  # This one should be routed on S1 and bridged on S2
    blue:                 # This one should be bridged everywhere
    green:                # and this one should be a regular VLAN interface on S1 and S2
  name: Simple trunk
#  disable: True
- sw:
  br:
  vlan.access: red        # This one should be routed on S1 and bridged on S2
  name: Access link to switch
#  disable: True
- sw:
  br:
  vlan.mode: route
  vlan.trunk:             # All three VLANs in the trunk should be routed on both ends due to link vlan.mode
    red:
    blue:
    green:
  name: Routed trunk
#  disable: True
- sw:
  br:
  vlan.access: red
  vlan.mode: route        # This one should be routed on both ends
#  disable: True
- sw:                     # Adding trunk between a switch and a bridge just for the fun of it ;)
  br:
  vlan.trunk:
    red:
    blue:
    green:
  name: Trunk between switch and bridge
#  disable: True
- rt:
  br:
  vlan.trunk:
    red:                  # This one should be routed on S1 and bridged on S3
    blue:                 # This one should be bridged everywhere
    green:                # and this one should be a regular VLAN interface on S1 and bridged on S3
  name: Trunk with bridge
#  disable: True
- rt:
  br:
  vlan.access: red        # This one should be routed on S1 and bridged on S3
  name: Access link to bridge
#  disable: True
