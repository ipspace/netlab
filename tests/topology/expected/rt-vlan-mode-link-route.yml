input:
- topology/input/rt-vlan-mode-link-route.yml
- package:topology-defaults.yml
links:
- bridge: input_1
  gateway:
    ipv4: 172.16.3.2/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.3.1/24
    node: h
  - _vlan_mode: route
    ifindex: 1
    ifname: eth1
    ipv4: 172.16.3.2/24
    node: rt
    vlan:
      access: red
      mode: route
  linkindex: 1
  name: Host access link
  node_count: 2
  prefix:
    ipv4: 172.16.3.0/24
  role: stub
  type: lan
  vlan:
    access: red
    mode: route
- bridge: input_2
  interfaces:
  - ifindex: 2
    ifname: eth2
    ipv4: 172.16.0.1/24
    node: h
  - _vlan_mode: bridge
    ifindex: 1
    ifname: eth1
    node: sw
    vlan:
      access: red
  linkindex: 2
  name: Host access link to a switch
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
  vlan:
    access: red
- bridge: input_3
  interfaces:
  - ifindex: 3
    ifname: eth3
    ipv4: 172.16.0.1/24
    node: h
  - _vlan_mode: bridge
    ifindex: 1
    ifname: eth1
    node: br
    vlan:
      access: red
  linkindex: 3
  name: Host access link to a bridge
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
  vlan:
    access: red
- interfaces:
  - ifindex: 2
    ifname: eth2
    node: sw
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 2
    ifname: eth2
    node: br
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 4
  name: Simple trunk
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    trunk:
      blue: {}
      green: {}
      red: {}
- bridge: input_5
  interfaces:
  - _vlan_mode: bridge
    ifindex: 3
    ifname: eth3
    node: sw
    vlan:
      access: red
  - _vlan_mode: bridge
    ifindex: 3
    ifname: eth3
    node: br
    vlan:
      access: red
  linkindex: 5
  name: Access link to switch
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
  vlan:
    access: red
- interfaces:
  - ifindex: 4
    ifname: eth4
    node: sw
    vlan:
      mode: route
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 4
    ifname: eth4
    node: br
    vlan:
      mode: route
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 6
  name: Routed trunk
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    mode: route
    trunk:
      blue: {}
      green: {}
      red: {}
- interfaces:
  - _vlan_mode: route
    ifindex: 5
    ifname: eth5
    ipv4: 10.1.0.2/30
    node: sw
    vlan:
      access: red
      mode: route
  - _vlan_mode: route
    ifindex: 5
    ifname: eth5
    ipv4: 10.1.0.1/30
    node: br
    vlan:
      access: red
      mode: route
  linkindex: 7
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
  vlan:
    access: red
    mode: route
- interfaces:
  - ifindex: 6
    ifname: eth6
    node: sw
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 6
    ifname: eth6
    node: br
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 8
  name: Trunk between switch and bridge
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    trunk:
      blue: {}
      green: {}
      red: {}
- interfaces:
  - ifindex: 2
    ifname: eth2
    node: rt
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  - ifindex: 7
    ifname: eth7
    node: br
    vlan:
      trunk:
        blue: {}
        green: {}
        red: {}
  linkindex: 9
  name: Trunk with bridge
  node_count: 2
  prefix: {}
  type: p2p
  vlan:
    trunk:
      blue: {}
      green: {}
      red: {}
- bridge: input_10
  interfaces:
  - _vlan_mode: route
    ifindex: 3
    ifname: eth3
    ipv4: 172.16.0.2/24
    node: rt
    vlan:
      access: red
  - _vlan_mode: bridge
    ifindex: 8
    ifname: eth8
    node: br
    vlan:
      access: red
  linkindex: 10
  name: Access link to bridge
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
  vlan:
    access: red
module:
- vlan
name: input
nodes:
  br:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: frr
    id: 4
    interfaces:
    - bridge: input_3
      ifindex: 1
      ifname: eth1
      linkindex: 3
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 2
      ifname: eth2
      linkindex: 4
      name: Simple trunk
      neighbors:
      - ifname: eth2
        node: sw
      subif_index: 3
      type: p2p
    - bridge: input_5
      ifindex: 3
      ifname: eth3
      linkindex: 5
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 4
      ifname: eth4
      linkindex: 6
      name: Routed trunk
      neighbors:
      - ifname: eth4
        node: sw
      subif_index: 3
      type: p2p
      vlan:
        mode: route
    - bridge_group: 1
      ifindex: 5
      ifname: eth5
      ipv4: 10.1.0.1/30
      linkindex: 7
      name: br -> sw
      neighbors:
      - ifname: eth5
        ipv4: 10.1.0.2/30
        node: sw
      type: p2p
      vlan:
        mode: route
    - ifindex: 6
      ifname: eth6
      linkindex: 8
      name: Trunk between switch and bridge
      neighbors:
      - ifname: eth6
        node: sw
      subif_index: 3
      type: p2p
    - ifindex: 7
      ifname: eth7
      linkindex: 9
      name: Trunk with bridge
      neighbors:
      - ifname: eth2
        node: rt
      subif_index: 3
      type: p2p
    - bridge: input_10
      ifindex: 8
      ifname: eth8
      linkindex: 10
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 9
      ifname: eth2.1001
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 10
      ifname: eth2.1002
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: green
        access_id: 1002
    - ifindex: 11
      ifname: eth2.1000
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - bridge_group: 2
      ifindex: 12
      ifname: eth4.1001
      ipv4: 172.16.4.4/24
      name: br -> sw
      neighbors:
      - ifname: eth4.1001
        ipv4: 172.16.4.3/24
        node: sw
      parent_ifindex: 4
      parent_ifname: eth4
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
        routed_link: true
    - bridge_group: 3
      ifindex: 13
      ifname: eth4.1002
      ipv4: 172.16.5.4/24
      name: br -> sw
      neighbors:
      - ifname: eth4.1002
        ipv4: 172.16.5.3/24
        node: sw
      parent_ifindex: 4
      parent_ifname: eth4
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1002
        mode: route
        routed_link: true
    - bridge_group: 1
      ifindex: 14
      ifname: eth4.1000
      ipv4: 172.16.6.4/24
      name: br -> sw
      neighbors:
      - ifname: eth4.1000
        ipv4: 172.16.6.3/24
        node: sw
      parent_ifindex: 4
      parent_ifname: eth4
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
        routed_link: true
    - ifindex: 15
      ifname: eth6.1001
      parent_ifindex: 6
      parent_ifname: eth6
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 16
      ifname: eth6.1002
      parent_ifindex: 6
      parent_ifname: eth6
      type: vlan_member
      virtual_interface: true
      vlan:
        access: green
        access_id: 1002
    - ifindex: 17
      ifname: eth6.1000
      parent_ifindex: 6
      parent_ifname: eth6
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - ifindex: 18
      ifname: eth7.1001
      parent_ifindex: 7
      parent_ifname: eth7
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 19
      ifname: eth7.1002
      parent_ifindex: 7
      parent_ifname: eth7
      type: vlan_member
      virtual_interface: true
      vlan:
        access: green
        access_id: 1002
    - ifindex: 20
      ifname: eth7.1000
      parent_ifindex: 7
      parent_ifname: eth7
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - bridge_group: 1
      ifindex: 21
      ifname: vlan1000
      name: VLAN red (1000) -> [rt,h,sw]
      neighbors:
      - ifname: eth2.1000
        ipv4: 172.16.0.2/24
        node: rt
      - ifname: eth2
        ipv4: 172.16.0.1/24
        node: h
      - ifname: vlan1000
        node: sw
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
    - bridge_group: 2
      ifindex: 22
      ifname: vlan1001
      name: VLAN blue (1001) -> [rt,sw]
      neighbors:
      - ifname: vlan1001
        node: rt
      - ifname: vlan1001
        node: sw
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
    - bridge_group: 3
      ifindex: 23
      ifname: vlan1002
      name: VLAN green (1002) -> [rt,sw]
      neighbors:
      - ifname: vlan1002
        ipv4: 172.16.2.2/24
        node: rt
      - ifname: vlan1002
        ipv4: 172.16.2.3/24
        node: sw
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.4/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.104
      mac: 08:4f:a9:00:00:04
    module:
    - vlan
    name: br
    vlan:
      max_bridge_group: 3
      mode: bridge
    vlans:
      blue:
        bridge_group: 2
        id: 1001
        mode: bridge
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
      green:
        bridge_group: 3
        id: 1002
        mode: bridge
        prefix:
          allocation: id_based
          ipv4: 172.16.2.0/24
      red:
        bridge_group: 1
        id: 1000
        mode: bridge
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
  h:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 1
    interfaces:
    - bridge: input_1
      gateway:
        ipv4: 172.16.3.2/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.3.1/24
      linkindex: 1
      name: Host access link
      neighbors:
      - ifname: eth1
        ipv4: 172.16.3.2/24
        node: rt
      role: stub
      type: lan
    - bridge: input_2
      gateway:
        ipv4: 172.16.0.2/24
      ifindex: 2
      ifname: eth2
      ipv4: 172.16.0.1/24
      linkindex: 2
      name: Host access link to a switch -> [rt,sw,br]
      neighbors:
      - ifname: eth2.1000
        ipv4: 172.16.0.2/24
        node: rt
      - ifname: vlan1000
        node: sw
      - ifname: vlan1000
        node: br
      type: lan
    - bridge: input_3
      gateway:
        ipv4: 172.16.0.2/24
      ifindex: 3
      ifname: eth3
      ipv4: 172.16.0.1/24
      linkindex: 3
      name: Host access link to a bridge -> [rt,sw,br]
      neighbors:
      - ifname: eth2.1000
        ipv4: 172.16.0.2/24
        node: rt
      - ifname: vlan1000
        node: sw
      - ifname: vlan1000
        node: br
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    name: h
    role: host
  rt:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: frr
    id: 2
    interfaces:
    - bridge: input_1
      bridge_group: 1
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.3.2/24
      linkindex: 1
      name: Host access link
      neighbors:
      - ifname: eth1
        ipv4: 172.16.3.1/24
        node: h
      role: stub
      type: lan
      vlan:
        mode: route
    - ifindex: 2
      ifname: eth2
      linkindex: 9
      name: Trunk with bridge
      neighbors:
      - ifname: eth7
        node: br
      subif_index: 3
      type: p2p
    - bridge: input_10
      bridge_group: 1
      ifindex: 3
      ifname: eth3
      ipv4: 172.16.0.2/24
      linkindex: 10
      name: Access link to bridge -> [h,sw,br]
      neighbors:
      - ifname: eth2
        ipv4: 172.16.0.1/24
        node: h
      - ifname: vlan1000
        node: sw
      - ifname: vlan1000
        node: br
      type: lan
      vlan:
        mode: route
    - ifindex: 4
      ifname: eth2.1001
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 5
      ifname: eth2.1002
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: green
        access_id: 1002
    - bridge_group: 1
      ifindex: 6
      ifname: eth2.1000
      ipv4: 172.16.0.2/24
      name: rt -> [h,sw,br]
      neighbors:
      - ifname: eth2
        ipv4: 172.16.0.1/24
        node: h
      - ifname: vlan1000
        node: sw
      - ifname: vlan1000
        node: br
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
    - bridge_group: 2
      ifindex: 7
      ifname: vlan1001
      name: VLAN blue (1001) -> [br,sw]
      neighbors:
      - ifname: vlan1001
        node: br
      - ifname: vlan1001
        node: sw
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
    - bridge_group: 3
      ifindex: 8
      ifname: vlan1002
      ipv4: 172.16.2.2/24
      name: VLAN green (1002) -> [br,sw]
      neighbors:
      - ifname: vlan1002
        node: br
      - ifname: vlan1002
        ipv4: 172.16.2.3/24
        node: sw
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.2/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - vlan
    name: rt
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 2
        id: 1001
        mode: bridge
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
      green:
        bridge_group: 3
        id: 1002
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.2.0/24
      red:
        bridge_group: 1
        id: 1000
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
  sw:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: frr
    id: 3
    interfaces:
    - bridge: input_2
      ifindex: 1
      ifname: eth1
      linkindex: 2
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 2
      ifname: eth2
      linkindex: 4
      name: Simple trunk
      neighbors:
      - ifname: eth2
        node: br
      subif_index: 3
      type: p2p
    - bridge: input_5
      ifindex: 3
      ifname: eth3
      linkindex: 5
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 4
      ifname: eth4
      linkindex: 6
      name: Routed trunk
      neighbors:
      - ifname: eth4
        node: br
      subif_index: 3
      type: p2p
      vlan:
        mode: route
    - bridge_group: 1
      ifindex: 5
      ifname: eth5
      ipv4: 10.1.0.2/30
      linkindex: 7
      name: sw -> br
      neighbors:
      - ifname: eth5
        ipv4: 10.1.0.1/30
        node: br
      type: p2p
      vlan:
        mode: route
    - ifindex: 6
      ifname: eth6
      linkindex: 8
      name: Trunk between switch and bridge
      neighbors:
      - ifname: eth6
        node: br
      subif_index: 3
      type: p2p
    - ifindex: 7
      ifname: eth2.1001
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 8
      ifname: eth2.1002
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: green
        access_id: 1002
    - ifindex: 9
      ifname: eth2.1000
      parent_ifindex: 2
      parent_ifname: eth2
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - bridge_group: 2
      ifindex: 10
      ifname: eth4.1001
      ipv4: 172.16.4.3/24
      name: sw -> br
      neighbors:
      - ifname: eth4.1001
        ipv4: 172.16.4.4/24
        node: br
      parent_ifindex: 4
      parent_ifname: eth4
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
        routed_link: true
    - bridge_group: 3
      ifindex: 11
      ifname: eth4.1002
      ipv4: 172.16.5.3/24
      name: sw -> br
      neighbors:
      - ifname: eth4.1002
        ipv4: 172.16.5.4/24
        node: br
      parent_ifindex: 4
      parent_ifname: eth4
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1002
        mode: route
        routed_link: true
    - bridge_group: 1
      ifindex: 12
      ifname: eth4.1000
      ipv4: 172.16.6.3/24
      name: sw -> br
      neighbors:
      - ifname: eth4.1000
        ipv4: 172.16.6.4/24
        node: br
      parent_ifindex: 4
      parent_ifname: eth4
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
        routed_link: true
    - ifindex: 13
      ifname: eth6.1001
      parent_ifindex: 6
      parent_ifname: eth6
      type: vlan_member
      virtual_interface: true
      vlan:
        access: blue
        access_id: 1001
    - ifindex: 14
      ifname: eth6.1002
      parent_ifindex: 6
      parent_ifname: eth6
      type: vlan_member
      virtual_interface: true
      vlan:
        access: green
        access_id: 1002
    - ifindex: 15
      ifname: eth6.1000
      parent_ifindex: 6
      parent_ifname: eth6
      type: vlan_member
      virtual_interface: true
      vlan:
        access: red
        access_id: 1000
    - bridge_group: 1
      ifindex: 16
      ifname: vlan1000
      name: VLAN red (1000) -> [rt,h,br]
      neighbors:
      - ifname: eth2.1000
        ipv4: 172.16.0.2/24
        node: rt
      - ifname: eth2
        ipv4: 172.16.0.1/24
        node: h
      - ifname: vlan1000
        node: br
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
    - bridge_group: 2
      ifindex: 17
      ifname: vlan1001
      name: VLAN blue (1001) -> [br,rt]
      neighbors:
      - ifname: vlan1001
        node: br
      - ifname: vlan1001
        node: rt
      type: svi
      virtual_interface: true
      vlan:
        mode: bridge
    - bridge_group: 3
      ifindex: 18
      ifname: vlan1002
      ipv4: 172.16.2.3/24
      name: VLAN green (1002) -> [br,rt]
      neighbors:
      - ifname: vlan1002
        node: br
      - ifname: vlan1002
        ipv4: 172.16.2.2/24
        node: rt
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.3/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    module:
    - vlan
    name: sw
    vlan:
      max_bridge_group: 3
    vlans:
      blue:
        bridge_group: 2
        id: 1001
        mode: bridge
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
      green:
        bridge_group: 3
        id: 1002
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.2.0/24
      red:
        bridge_group: 1
        id: 1000
        mode: bridge
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
provider: libvirt
vlans:
  blue:
    host_count: 0
    id: 1001
    mode: bridge
    neighbors:
    - ifname: vlan1001
      node: br
    - ifname: vlan1001
      node: rt
    - ifname: vlan1001
      node: sw
    prefix:
      allocation: id_based
      ipv4: 172.16.1.0/24
  green:
    host_count: 0
    id: 1002
    neighbors:
    - ifname: vlan1002
      node: br
    - ifname: vlan1002
      ipv4: 172.16.2.2/24
      node: rt
    - ifname: vlan1002
      ipv4: 172.16.2.3/24
      node: sw
    prefix:
      allocation: id_based
      ipv4: 172.16.2.0/24
  red:
    host_count: 1
    id: 1000
    mode: route
    neighbors:
    - ifname: eth2.1000
      ipv4: 172.16.0.2/24
      node: rt
    - ifname: eth2
      ipv4: 172.16.0.1/24
      node: h
    - ifname: vlan1000
      node: sw
    - ifname: vlan1000
      node: br
    prefix:
      allocation: id_based
      ipv4: 172.16.0.0/24
